packages/front-end/components/Settings/UpgradeModal/index.tsx

```diff
@@ -0,0 +1,44 @@
+import Modal from "../../Modal";
+import { useUser } from "../../../services/UserContext";
+import CloudUpgradeForm from "./CloudUpgradeForm";
+import SelfHostedUpgradeForm from "./SelfHostedUpgradeForm";
+import { useState } from "react";
+
+export const currencyFormatter = new Intl.NumberFormat(undefined, {
+  style: "currency",
+  currency: "USD",
+});
+
+export interface Props {
+  close: () => void;
+  source: string;
+  reason: string;
+}
+
+export default function UpgradeModal({ close, source, reason }: Props) {
+  const [closeCta, setCloseCta] = useState("Cancel");
+  const { accountPlan } = useUser();
+  if (["pro", "pro_sso", "enterprise"].includes(accountPlan)) {
+    close();
+  }
+
+  return (
+    <Modal open={true} close={close} closeCta={closeCta} size="lg">
+      {accountPlan === "starter" ? (
```

Comment:
Use `isCloud()` for this instead.

Reply:
this component should actually be deleted in favor of `components/Marketing/UpgradeMessage.tsx`, which does switch on `isCloud()`